### ISSUE TITLE
Rule to warn against shareReplay({ refCount: false }) before takeUntil

### ISSUE THREAD

**Issue by JasonWeinzierl** [JasonWeinzierl/eslint-plugin-rxjs-x#108]
Putting `shareReplay` before `takeUntil` with `refCount: false` may cause unexpected behavior because the source observable will not be unsubscribed.  See the "An update" section in this blog post: https://ncjamieson.com/avoiding-takeuntil-leaks/#an-update .  We should implement a new rule that errors on this case.

```ts
import { interval, shareReplay, takeUntil } from 'rxjs';

const destroy = new Subject<void>();
const ticker = interval(1000).pipe(
  tap(value => console.log(value)),
  shareReplay({ refCount: false }),
  takeUntil(destroy),
).subscribe();

destroy.next();

// Despite the takeUntil causing an unsubscribe, the shareReplay remains open and the interval will keep going.
```

**Comment by JasonWeinzierl** [comment]
New rule `no-sharereplay-before-takeuntil` is released in v0.8.1
